abi/src/main/java/org/web3j/abi/datatypes/DynamicArray.java

```diff
@@ -66,7 +66,7 @@ public DynamicArray(Class<T> type, T... values) {
     @Override
     public String getTypeAsString() {
         String type;
-        if (StructType.class.isAssignableFrom(value.get(0).getClass())) {
+        if (!value.isEmpty() && StructType.class.isAssignableFrom(value.get(0).getClass())) {
             type = value.get(0).getTypeAsString();
         } else {
             type = AbiTypes.getTypeAString(getComponentType());
```

Comment:
This throws a `NullPointerException` if I'm not wrong?

Reply:
It can throw IndexOutOfBoundsException while accessing empty array